training/src/main/java/wolox/training/services/OpenLibraryService.java

```diff
@@ -0,0 +1,79 @@
+package wolox.training.services;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+import wolox.training.models.Book;
+import wolox.training.models.DTO.LibraryServiceDTO;
+
+@Service
+public class OpenLibraryService {
+
+    private final RestTemplate restTemplate;
+
+    public OpenLibraryService(RestTemplateBuilder restTemplateBuilder) {
+        this.restTemplate = restTemplateBuilder.build();
+    }
+
+    private String getNumberOfAlphanumericString(String alphanumericString) {
+        Pattern regex = Pattern.compile("\\d+");
+        Matcher numberFound = regex.matcher(alphanumericString);
+        while (numberFound.find()) {
+            return numberFound.group(0);
+        }
+        return "0";
+    }
+
+
+    public Optional<Book> bookInfo(String isbn) throws ParseException, JsonProcessingException {
+        Book bookToReturn = null;
+        JSONParser parser = new JSONParser();
+        ObjectMapper objectMapper = new ObjectMapper();
+        LibraryServiceDTO dtoLibraryService = new LibraryServiceDTO();
+
+        String url = "https://openlibrary.org/api/books?bibkeys={isbn}&format=json&jscmd=data";
+        //make a request
+        ResponseEntity<String> response = this.restTemplate
+            .exchange(url, HttpMethod.GET, null, String.class, isbn);
+        //check status ok
+        if (response.getStatusCode() == HttpStatus.OK && !response.getBody().equals("{}")) {
+            bookToReturn = new Book();
+            // parse response to Json
+            JSONObject json = (JSONObject) parser.parse(response.getBody());
+
+            //read json into tree model
+            JsonNode rootNode = objectMapper.readTree(String.valueOf(json)).get(isbn);
+
+            //map json To DTO
+            dtoLibraryService.mapJsonToDto(
+                rootNode.get("subjects"),
+                rootNode.get("authors"),
+                rootNode.get("cover").get("small").asText(),
+                rootNode.get("title").asText(),
+                rootNode.get("subtitle").asText(),
+                rootNode.get("publishers"),
+                rootNode.get("publish_date").asText(),
+                Integer.valueOf(rootNode.get("number_of_pages").asText()),
+                getNumberOfAlphanumericString(isbn)
```

Comment:
Wouldn't it be easier to pass the root node and let the method do all this logic?